src/include/binder/query/query_graph_label_analyzer.h

```diff
@@ -11,10 +11,10 @@ class QueryGraphLabelAnalyzer {
     explicit QueryGraphLabelAnalyzer(const main::ClientContext& clientContext, bool throwOnViolate)
         : throwOnViolate{throwOnViolate}, clientContext{clientContext} {}
 
-    void pruneLabel(QueryGraph& graph);
+    void pruneLabel(const QueryGraph& graph) const;
```

Comment:
Honestly this graph should not be const because are pruning labels out of it.

It works with const because we are modifying field of its field.

Reply:
Yeah. sometimes the clang tidy can be annoying as opposed to the semantics. I reverted the change and suppress it with 
```cpp
// NOLINTNEXTLINE(readability-non-const-parameter): graph is supposed to be modified.
```